### PR TITLE
Remove usage of *_STORAGE_API_ENDPOINT` environment variables for providers

### PR DESCRIPTION
**Pull request by shreyas-s-rao**
**How to categorize this PR?**

/area security
/kind task

**What this PR does / why we need it**:
Remove usage of *_STORAGE_API_ENDPOINT` environment variables for providers, since credentials from secrets (including API endpoints) should not be passed as environment variables to the backup-restore container.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/etcd-backup-restore/issues/727

**Special notes for your reviewer**:
/block
To be merged only after https://github.com/gardener/etcd-backup-restore/pull/759 is merged, released and etcdbr version v0.30.0 is updated in druid.
For local testing purpose, I have included the PR build image from https://concourse.ci.gardener.cloud/teams/gardener/pipelines/etcd-backup-restore-master/jobs/master-pull-request-job/builds/37. This commit will be removed before merging. 

@renormalize please let me know if we can pass `domain` to Azurite (with format `http(s)://<storageAccount>.<domain>`) instead of the entire URL.

**Release note**:

```noteworthy user
Remove usage of *_STORAGE_API_ENDPOINT` environment variables for Google and Azure providers. Storage API endpoint / domain will instead be directly consumed by etcd-backup-restore from the mounted backup secret.
```

### CODE REVIEW

File: internal/images/images.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ images:
     name: 'etcdbrctl'
   sourceRepository: github.com/gardener/etcd-backup-restore
   repository: europe-docker.pkg.dev/gardener-project/public/gardener/etcdbrctl
-  tag: "v0.29.0"
+  tag: "v0.30.0-dev-00e7d52969652e0390590750da06582ce6c336f5"
```
Comment (renormalize) [inline internal/images/images.yaml:17]: Could you revert this? Thanks.

File: test/e2e/utils.go [hunk 2]
```diff
@@ -370,6 +370,7 @@ func getProviders() ([]TestProvider, error) {
 				azuriteHost := getEnvOrFallback("AZURITE_HOST", "")
 				if azuriteHost != "" {
 					provider.Storage.SecretData["emulatorEnabled"] = []byte("true")
+					// TODO: replace with `domain`
```
Comment (renormalize) [inline test/e2e/utils.go:373]: Can you address this TODO? Thanks.

File: config/samples/etcd-secret-azurite.yaml [hunk 3]
```diff
@@ -10,4 +10,5 @@ data:
   storageAccount: ZGV2c3RvcmVhY2NvdW50MQ==
   storageKey: RWJ5OHZkTTAyeE5PY3FGbHFVd0pQTGxtRXRsQ0RYSjFPVXpGVDUwdVNSWjZJRnN1RnEyVVZFckN6NEk2dHEvSzFTWkZQVE90ci9LQkhCZWtzb0dNR3c9PQ==
   emulatorEnabled: dHJ1ZQ== # true
+  # TODO: replace with `domain`
```
Comment (renormalize) [inline config/samples/etcd-secret-azurite.yaml:13]: Can you address this TODO? Thanks.

**Review by renormalize — CHANGES_REQUESTED** [review]
Thanks for the PR @shreyas-s-rao!
Could you address the few TODOs that are remaining?

**Comment by renormalize** [comment]
As discussed off GitHub with @shreyas-s-rao, passing domain in the `http(s)://<storageAccount>.<domain>` format is supported by Azurite.

The way to perform this is to add:

```
127.0.0.1 account1.blob.localhost
```

to the hosts file of the host machine where etcd-backup-restore tests are running so that `https://<storage-account>.blob.localhost` resolves to the Azurite process. One simply needs to pass the storage account name that is configured to be used for Azurite, and the `domain` as `blob.localhost`.

However, it has to be kept in mind to *not* pass the `disableProductStyleUrl` flag to Azurite at startup. More can be found [here](https://github.com/Azure/Azurite?tab=readme-ov-file#production-style-url).

Doing this will break the current way the Azurite make targets work in etcd-druid, so for a few commits Azurite will not be functional with etcd-druid for local development and testing.

**Comment by shreyas-s-rao** [comment]
Added this as a point in #782

**Comment by ashwani2k** [comment]
/unhold as https://github.com/gardener/etcd-backup-restore/pull/759 is merged. 
@shreyas-s-rao  Shall we merge it?

**Comment by shreyas-s-rao** [comment]
@ashwani2k I've removed the temp commit now, so the PR can be merged once the tests pass.

**Comment by shreyas-s-rao** [comment]
@ashwani2k can you please raise a PR to update etcdbr image to v0.30.0 in images.yaml, along with the relevant release notes? I'm surprised that the PR was not raised automatically.

**Comment by shreyas-s-rao** [comment]
@ashwani2k I have rebased the PR, so it can now be merged, after the tests pass of course.

**Review by ashwani2k — APPROVED** [review]

**Review by renormalize — APPROVED** [review]
Tested with the `v0.30.1` of etcd-backup-restore, and all behavior is as expected. Thanks.

**Pull request merged**